client/core/trade.go

```diff
@@ -1223,6 +1231,14 @@ func (t *trackedTrade) finalizeSwapAction(match *matchTracker, coinID, contract
 	// penalized).
 	timeout := t.broadcastTimeout()
 	if err := t.dc.signAndRequest(init, msgjson.InitRoute, ack, timeout); err != nil {
+		var msgErr *msgjson.Error
+		if errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError {
+			// Try resolving the match status conflict.
+			go c.resolveMatchConflicts(t.dc, []*matchStatusConflict{{
+				trade: t,
+				match: match,
+			}})
+		}
```

Comment:
I actually don't think that we're likely to recover from this, so the call to `resolveMatchConflicts` is sort of a hail mary last ditch attempt to try to make sense of the situation.